### PR TITLE
Feature/incremental update

### PR DESCRIPTION
**Pull request by Aurora12**
Incremental chats update as per https://www.notion.so/worldr/API-Incremental-update-on-launch-dbcedf7d56bb45ddafd741ed136395a3

Store tests only.

App layer tests and integration tests will follow in the next two PRs as there is a lot of stuff in there.

### CODE REVIEW

File: app/post.go [hunk 1]
```diff
@@ -1391,38 +1393,150 @@ func (a *App) GetRecentPosts(request *model.RecentPostsRequestData) (*model.Post
...
 		}
 	}
 	return count+missingCount <= MAX_INCREMENT_TOTAL, nil
 }
+
+// GetIncrementalPosts() returns a list of posts for given channels after the given last post id
+// for each channel. The response is paginated and the method respects the limit of maximum
+// messages per page. The client can provide a desired page size no more than MAX_INCREMENT_PAGE.
+//
+func (a *App) GetIncrementalPosts(request *model.IncrementPostsRequest) (*[]model.IncrementalPosts, *model.AppError) {
```
Comment (ygworldr) [inline app/post.go:1450]: Long function but well documented so that is fine.

File: app/post.go [hunk 2]
```diff
@@ -1391,38 +1393,150 @@ func (a *App) GetRecentPosts(request *model.RecentPostsRequestData) (*model.Post
...
+		for i := range *posts {
+			result = append(result, &((*posts)[i]))
+		}
+	}
+	return &result
+}
+
+// CheckIncrementPossible() returns true if it is reasonable to proceed with downloading incremental updates for given chats.
+//
 func (a *App) CheckIncrementPossible(request *model.IncrementCheckRequest) (bool, *model.AppError) {
```
[the remark this answers is not in the document]
Comment (Aurora12) [reply app/post.go:1420]: The comment is just above, line 1418. Not much to say about it I think. Do you want me to elaborate on how the method does its work here?

[the remark this answers is not in the document]
Comment (ygworldr) [reply app/post.go:1420]: Err… No, it is fine. My mistake!

File: Makefile [hunk 3]
```diff
@@ -217,28 +217,42 @@ i18n-extract: ## Extract strings for translation from the source code
 	$(GOBIN)/mmgotool i18n extract
 
 store-mocks: ## Creates mock files.
-	env GO111MODULE=off $(GO) get -u github.com/vektra/mockery/...
+	# FIXME: the next line uses a deprecated method of getting mockery library
```
[the remark this answers is not in the document]
Comment (ygworldr) [reply Makefile:220]: This is the only thing really.

[the remark this answers is not in the document]
Comment (Aurora12) [reply Makefile]: I've changed the comment as per discussion in TG. It now says:

	# The mock code is generated by mockery library.
	# Get the binary at https://github.com/vektra/mockery/releases/tag/v1.1.0
	# and place it in server/bin.

We may choose to add automatics here later, but that is not important for now.

[the remark this answers is not in the document]
Comment (ygworldr) [reply Makefile]: :+1:

**Pull request merged**